### PR TITLE
feat: implement HTML DOCTYPE lexer matching SWC spec

### PR DESCRIPTION
**Pull request by aryanraj45**
**Summary**

This PR implements the full WHATWG-compliant DOCTYPE state machine in the HTML Tokenizer (`walkHtmlTokens.js`), directly mirroring the reference implementation in the [SWC HTML parser](https://github.com/swc-project/swc/blob/main/crates/swc_html_parser/src/lexer/mod.rs).
Previously, the tokenizer took a shortcut by dumping `<!DOCTYPE>` elements into a generic fallback state and emitting them as standard comments. This PR resolves that architectural mismatch and fully complies with the WHATWG HTML5 parsing specification by:

1. **Implementing all 16 DOCTYPE States**: Ported the complete state machine (e.g., `BeforeDoctypeName`, `DoctypePublicIdentifierDoubleQuoted`, etc.) to properly parse DOCTYPE internal structures and EOF edge cases.
2. **Dedicated Token Emission**: The tokenizer now natively emits DOCTYPEs via a dedicated `callbacks.doctype()` event, rather than conflating them with `callbacks.comment()`.
3. **Zero-Allocation Maintained**: The new states strictly adhere to Webpack's performance requirements, utilizing fast integer character code (`cc`) comparisons and direct index tracking without expensive string allocations.
4. **Spec Traceability**: Original WHATWG specification URL comments have been included above every state block to match the SWC source and aid future maintainability.

Supports #536 

**What kind of change does this PR introduce?**

feat

**Did you add tests for your changes?**

yes, updated `test/walkHtmlTokens.unittest.js` to assert the new `doctype` callbacks, handle DOCTYPE EOF edge cases, verify case-insensitivity, and updated the token snapshots. All 4,600+ `ConfigTestCases` pass successfully.

**Does this PR introduce a breaking change?**

no

**If relevant, what needs to be documented once your changes are merged or what have you already documented?**

n/a

**Use of AI**

Partial

### CODE REVIEW

File: .changeset/html-tokenizer-doctype.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"webpack": patch
+---
+
+Implement full WHATWG-compliant state machine for DOCTYPEs in HTML tokenizer
```
Comment (alexander-akait) [inline .changeset/html-tokenizer-doctype.md]: Avoid it, because it is not publicly changes

File: lib/html/walkHtmlTokens.js [hunk 2]
```diff
@@ -686,15 +748,765 @@ const walkHtmlTokens = (input, pos = 0, callbacks = {}) => {
...
+					// Append the current input character to the comment token's data.
+					pos++;
+				} else {
+					// Anything else
+					// Reconsume in the comment state.
+					state = STATE_COMMENT;
+					// Reconsume
+				}
+				break;
+
```
[the remark this answers is not in the document]
Comment (alexander-akait) [reply lib/html/walkHtmlTokens.js:777]: Let's fix it or put todo about future improvements and where

**Comment by aryanraj45** [comment]
@alexander-akait done applied the fix and fixed the lint issue please take a look thanks :))

File: lib/html/walkHtmlTokens.js [hunk 3]
```diff
@@ -774,7 +792,10 @@ const walkHtmlTokens = (input, pos = 0, callbacks = {}) => {
 				// Consume the next input character:
 				// U+002D HYPHEN-MINUS (-)
 				// Switch to the comment less-than sign bang dash state.
-				if (cc === CC_HYPHEN_MINUS) {
+				if (cc === CC_LESS_THAN) {
```
Comment (alexander-akait) [inline lib/html/walkHtmlTokens.js]: Comments and logic are different please port 1:1 code, don't modify something

**Review by alexander-akait — CHANGES_REQUESTED** [review]
Please migrate code 1:1, comments are about `HYPHEN-MINUS`, but you use `CC_LESS_THAN`

**Comment by aryanraj45** [comment]
@alexander-akait done  fixed the mismatch in the comment lexer the code logic and spec comments are now perfectly aligned.can u please take a look thanks !!

**Comment by aryanraj45** [comment]
@alexander-akait all the required CI passes

**Review by alexander-akait — APPROVED** [review]
